cloudsmith_cli/cli/commands/upstream.py

```diff
@@ -0,0 +1,449 @@
+# -*- coding: utf-8 -*-
+"""CLI/Commands - create, retrieve, update or delete repository upstreams."""
+from __future__ import absolute_import, print_function, unicode_literals
+
+import json
+
+import click
+import six
+
+from ...core.api import upstreams as api
+from .. import command, decorators, utils, validators
+from ..exceptions import handle_api_exceptions
+from ..utils import (
+    fmt_bool,
+    fmt_datetime,
+    maybe_spinner,
+    maybe_truncate_list,
+    maybe_truncate_string,
+)
+from .main import main
+
+UPSTREAM_FORMATS = [
```

Comment:
NAB: it's a shame we can't leverage the existing formats API to get a list of supported upstream formats. Perhaps enriching it with an attribute for `supports_upstream`, or query could be neat. This approach does work of course, but ties upstream format releases to a new CLI release.

For example, here we filter packages that use distros: https://github.com/cloudsmith-io/cloudsmith-cli/blob/5933eb1bf09f1d322d51f0112aae223e37116f1c/cloudsmith_cli/core/api/packages.py#L292-L294

Reply:
True. Sounds like a ticket for QOL pile.